### PR TITLE
Enable Path, File and Directory Assert/Constraints in the .NET Standard Build

### LINKED ISSUE #1904: Add .NET Standard 1.6 Dependencies to the Nuspec Files
I just noticed that I didn't add a dependency on the .NET Standard libraries to the Nuspec files for the .NET Standard version of NUnit and NUnitLite.

Probably something like,

```xml
    <dependencies>
      <group targetFramework="netstandard1.6">
        <dependency id="NETStandard.Library" version="1.6.0" />
      </group>
    </dependencies>
```

### PR DESCRIPTION
**Pull request by rprouse**
This started off as just dropping the PORTABLE define from the .NET Standard builds so that we can start to re-enable some of the features that are not available in the PORTABLE version of the framework. As a part of that, I have documented what is still not enabled and will create issues to document or re-enable those that are remaining.

As a part of this, I enabled all of the file, path and directory based asserts that weren't available in PORTABLE. I don't have an issue for that, so I am marking this PR with the 3.6 milestone to track for the release.

The steps I took for this were,

1. Remove the PORTABLE define in all of the .NET Standard projects
2. Wherever possible, rewrite code that wasn't compiling to work in .NET Standard
3. As a last resort, add the needed `#if NETSTANDARD1_6` blocks and document the not supported functionality

This is a large PR because so many files were touched, but most of the changes are fairly rote.

I also updated the nuspec files which fixes #1904

### CODE REVIEW

File: src/NUnitFramework/nunitlite-runner/nunitlite-runner-netstandard.csproj [hunk 1]
```diff
@@ -49,12 +49,6 @@
     <Compile Include="..\FrameworkVersion.cs">
       <Link>Properties\FrameworkVersion.cs</Link>
     </Compile>
-    <Compile Include="..\nunitlite\ColorConsole.cs">
```
Comment (rprouse) [inline src/NUnitFramework/nunitlite-runner/nunitlite-runner-netstandard.csproj]: These are now included in NUnitLite like in the other projects.

File: src/NUnitFramework/nunitlite/Options.cs [hunk 2]
```diff
@@ -539,23 +539,6 @@ public override void GetObjectData (SerializationInfo info, StreamingContext con
 
     public class OptionSet : KeyedCollection<string, Option>
     {
-#if !PORTABLE
```
Comment (rprouse) [inline src/NUnitFramework/nunitlite/Options.cs]: We never use the localizer, so easier to easier to just use the PORTABLE null-op version for everything.

**Comment by rprouse** [comment]
I have switched all of the async code to use `#if ASYNC` which is more explicit, like we do with `#if PARALLEL`

**Comment by CharliePoole** [comment]
It looks good to me but I think you may want to merge in the latest from master manually. I believe I made some changes to defines that you don't have. I'm OK either way though.

**Comment by rprouse** [comment]
I will pull in latest from master and make sure everything builds and tests pass before merging.

**Comment by rprouse** [comment]
I've merge latest from master and ran all tests which are passing locally. This can be merged once CI passes.

**Comment by CharliePoole** [comment]
I cancelled the Travis build so the netstandard branch could run. Everything was passing but it couldn't seem to find any Apple agents to run on.

**Comment by CharliePoole** [comment]
Travis doesn't seem to be able to find Apple agents tonight, but the Linux jobs all passed so I'm going to go ahead and merge.

**Pull request merged**